### ISSUE TITLE
Delay the visibility of the spinner

### ISSUE THREAD

**Issue by mpalourdio** [mpalourdio/ng-http-loader#26]
It can be annoying to have the spinner shown for very short requests. Even if there's possibility to exclude some url patterns, a debounce time should be configurable before showning the spinner.

Hiding the spinner should not be affected by the debounce time and should be immediate. It's important for ``forkJoin`` observables.

Feature request here : https://github.com/mpalourdio/ng-http-loader/issues/25

**Comment by mpalourdio** [comment]
```typescript
export class SpinnerComponent implements OnDestroy, OnInit {
    [...]
    public debounceDelay = 0;
    constructor(private pendingRequestInterceptorService: PendingInterceptorService) {
        this.subscription = this.pendingRequestInterceptorService
            .pendingRequestsStatus
            .debounce(val => {
                if (val) {
                    return timer(this.debounceDelay);
                } else {
                    return timer(0); //do not delay hiding
                }
            })
            .subscribe(hasPendingRequests => {
                this.isSpinnerVisible = hasPendingRequests;
            });
    }
}
```

**Comment by mpalourdio** [comment]
Fixed by #27
